x-py.md

```diff
@@ -0,0 +1,388 @@
+        --no-fail-fast  Run all tests regardless of failure
+        --test-args ARGS
+                        extra arguments
+```
+
+The last two flags `--no-fail-fast` and `--test-args` are the extra flags
+available for `./x.py test` compared to `build`.
+
+# Benchmarking rustc
+
+This one is a easier compared to the others. All you're doing is
+running benchmarks of the compiler itself so it'll build it and run the one set
+of benchmarks available to it. The command is:
+
+```bash
+./x.py bench
+```
+
+### Other Flags
+
+The `bench` option does have flags available to it and they're shown below.
+They're the same almost as `test` and `build` but it lacks the `--no-fail-fast`
+flag that `test` has.
+
+```
+Options:
+    -v, --verbose       use verbose output (-vv for very verbose)
+    -i, --incremental   use incremental compilation
+        --config FILE   TOML configuration file for build
+        --build BUILD   build target of the stage0 compiler
+        --host HOST     host targets to build
+        --target TARGET target targets to build
+        --on-fail CMD   command to run on failure
+        --stage N       stage to build
+        --keep-stage N  stage to keep without recompiling
+        --src DIR       path to the root of the rust checkout
+    -j, --jobs JOBS     number of jobs to run in parallel
+    -h, --help          print this help message
+        --test-args ARGS
+                        extra arguments
+```
+
+# Documenting rustc
+
+You might want to build documentation of the various components available like
+the standard library. There's two ways to go about this. You can run `rustdoc`
+directly on the file to make sure the HTML is correct which is fast or you can
+build the documentation as part of the build process through `x.py`. Both are
+viable methods since documentation is more about the content.
+
+### Document everything
+
+```bash
+# Document it all
+./x.py doc
+
+# If you want to avoid the whole Stage 2 build
+./x.py doc --stage 1
+```
+
+First the compiler and rustdoc get built to make sure everything is okay and
+then it documents the files.
+
+### Document specific components
+
+```bash
+./x.py doc src/doc/book
+./x.py doc src/doc/nomicon
+./x.py doc src/doc/book src/libstd
+```
+
+Much like individual tests or building certain components you can build only the
+documentation you want.
+
+### Document internal rustc items
+
+By default `rustc` does not build the compiler for it's internal items.
+Mostly because this is useless for the average user. However, you might need to
+have it available so you can understand the types. Here's how you can compile it
+yourself:
+
+```bash
+cp src/bootstrap/config.toml.example config.toml
+```
+
+Next open up `config.toml` and make sure these two lines are set to true:
+
+```
+docs = true
+compiler-docs = true
+```
+
+When you want to build the compiler docs as well run this command:
+
+```bash
+./x.py doc --config config.toml
```

Comment:
It's not actually necessary in most cases to specify this, rustbuild will find the configuration on its own.